test/ia-account-settings.test.ts

```diff
@@ -0,0 +1,132 @@
+import { html, fixture, expect } from '@open-wc/testing';
+import { stub } from 'sinon';
+import type { IAAccountSettings } from '../src/ia-account-settings';
+import '../src/ia-account-settings';
+import MailingLists from '../src/assets/config';
+
+describe('IAAccountSettings', () => {
+  it('default user data/fields rendered on page', async () => {
+    const el = await fixture<IAAccountSettings>(
+      html`<ia-account-settings></ia-account-settings>`
+    );
+    el.userData = {
+      identifier: '@foo-bar',
+      screenname: 'foo-bar',
+      email: 'abc@gmail.com',
+      isAdmin: true,
+    };
+    el.mailingLists = MailingLists;
+    el.selectedMailingLists = ['ml_best_of', 'ml_donors'];
+    el.lookingToAuth = false;
+
+    await el.updateComplete;
+
+    const settingsTemplate = el.shadowRoot?.querySelector('.settings-template');
+
+    // has rendered initial fields for user
+    expect(settingsTemplate?.querySelector('ia-pic-uploader')).to.be.exist;
+    expect(settingsTemplate?.querySelector('#screenname')).to.be.exist;
+    expect(settingsTemplate?.querySelector('#email')).to.be.exist;
+    expect(settingsTemplate?.querySelector('#password')).to.be.exist;
+    expect(settingsTemplate?.querySelector('#borrow-history')).to.be.exist;
+
+    // you are an admin
+    expect(settingsTemplate?.querySelector('.admin-functions')).to.be.exist;
+
+    // has not linked with third-party accounts
+    expect(settingsTemplate?.querySelector('#linked-account')).to.not.be.exist;
+  });
+
+  it('display verification template', async () => {
+    const el = await fixture<IAAccountSettings>(
+      html`<ia-account-settings></ia-account-settings>`
+    );
+    el.userData = {
+      identifier: '@foo-bar',
+      screenname: 'foo-bar',
+      email: 'abc@gmail.com',
+      isAdmin: true,
+    };
+    el.mailingLists = MailingLists;
+    el.selectedMailingLists = ['ml_best_of', 'ml_donors'];
```

Comment:
Q: I'm a bit confused about the type of `selectedMailingLists`. Where it's defined in the models, it looks like it should be a key-value dict. But here it's being set to an array. Could this be made more consistent?

Reply:
I have changed the property types